include/boost/math/special_functions/detail/ibeta_inverse.hpp

```diff
@@ -25,23 +25,16 @@ namespace boost{ namespace math{ namespace detail{
 template <class T>
 struct temme_root_finder
 {
-   temme_root_finder(const T t_, const T a_) : t(t_), a(a_) {}
+   temme_root_finder(const T t_, const T a_) : t(t_), a(a_) {
+      const T x_extrema = 1 / (1 + a);
+      BOOST_MATH_ASSERT(0 < x_extrema && x_extrema < 1);
+   }
 
    boost::math::tuple<T, T> operator()(T x)
    {
       BOOST_MATH_STD_USING // ADL of std names
 
       T y = 1 - x;
-      if(y == 0)
```

Comment:
I'm not sure we can guarantee that this code is no longer needed?

It's purpose was to return "sane" values that the root finder can still use, even in the case that T has no infinity causing the log functions below to "explode".

Reply:
The purpose of this code appeared to be to avoid domain errors in the `log` functions when evaluating `x` values less than 0 or greater than `. In #1002, the upper Kolmogorov quantile search bound was changed to 1 from `boost::math::tools::max_value<RealType>()` so it doesn't appear to be necessary to be. Removing it doesn't appear to cause test failures.

Reply:
Trivial test case is `ibetac_inv(1e50, 10, 1./10)` which is from test_ibeta_inv.hpp:284 and does cause the test to fail here with msvc.

Reply:
This case never makes it to the Newton Solver because as it gets special cased out. Are you sure the branch you are testing is up to date?

https://github.com/boostorg/math/blob/e682c1be3326a7b4ead43ff1c50149a4ce226e2e/include/boost/math/special_functions/detail/ibeta_inverse.hpp#L407-L408

Reply:
You're right - my bad!